### PR TITLE
Added login page with authentication 

### PR DESCRIPTION
**Pull request by Shah-Arsalan**
Used a mock user from data base to complete login authentication . 
The code can be found in loginPage.jsx and auth-context.jsx

### CODE REVIEW

File: src/WebPages/loginPage/LoginPage.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      setTimeout(() => {
+        navigate("/");
+      }, 1000);
+    }
+  }, [token]);
+
+  return (
+    <>
+      <div className="login-page">
+        <div class="login-container">
```
Comment (ravikumar1002) [inline src/WebPages/loginPage/LoginPage.jsx]: you forgot to change it in className attribute

File: src/WebPages/loginPage/LoginPage.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                onClick={() =>
+                  loginCall(loginDetails.email, loginDetails.password)
+                }
+              >
+                Log In
+              </button>
+            </div>
+
+            <div class="input flex-justify-center">
+              <a href="/WebPages/signup/signup.html">
```
Comment (ravikumar1002) [inline src/WebPages/loginPage/LoginPage.jsx]: use Link  (React Router) tag

**Review by ravikumar1002 — APPROVED** [review]
everything looks good

File: src/App.js [hunk 3]
```diff
@@ -5,16 +5,21 @@ import { Routes, Route } from "react-router-dom";
 import { Nav } from "./Components/Nav/Nav.jsx";
 import { LandingPage } from "./WebPages/LandingPage/LandingPage";
 import { ProductListing } from "./WebPages/ProductListing/ProductListing";
-import { useData } from "./Contexts/DataContext";
+import { LoginPage } from "./WebPages/loginPage/LoginPage";
+import { useAuth } from "./Contexts/auth-context";
 
 function App() {
+  const { token, user } = useAuth();
+  console.log("token is", token);
```
Comment (Rohanmond) [inline src/App.js]: remove these consoles

File: src/App.js [hunk 4]
```diff
@@ -5,16 +5,21 @@ import { Routes, Route } from "react-router-dom";
...
+  console.log("activeUser is ", user);
```
Comment (Rohanmond) [inline src/App.js]: removes these also

File: src/Contexts/auth-context.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import axios from "axios";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const localStorageItems = JSON.parse(
+    localStorage.getItem("LoginCredentials")
+  );
+  const [token, setToken] = useState(localStorageItems?.userToken);
+  const [user, setUser] = useState(localStorageItems?.activeUser);
```
Comment (Rohanmond) [inline src/Contexts/auth-context.jsx:12]: liked the usage of optional chaining

File: src/Contexts/auth-context.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            userToken: response.data.encodedToken,
+            activeUser: response.data.foundUser,
+          })
+        );
+        setToken(response.data.encodedToken);
+        setUser(response.data.foundUser);
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
```
Comment (Rohanmond) [inline src/Contexts/auth-context.jsx]: you can remove this final block if you are not writing anything

**Review by Rohanmond — APPROVED** [review]

**Pull request merged**